### PR TITLE
Feature/nmd 468 create a function that allows achievements to be added to the user when their requirements are met

### PR DESCRIPTION
**Pull request by MaliyOna**
I add tests, because my pipeline won`t work without them

### CODE REVIEW

File: CatNote.DAL/Repositories/TaskRepository.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using CatNote.DAL.Entities;
```
Comment (kalachck) [inline CatNote.DAL/Repositories/TaskRepository.cs:6]: Unnecessary usings

File: CatNote.Domain/Enums/AchievementType.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿namespace CatNote.Domain.Enums;
+
+public enum AchievementType
+{
+    None,
```
Comment (kalachck) [inline CatNote.Domain/Enums/AchievementType.cs:5]: Use explicit Enums

File: CatNote.IntegrationTests/TaskTests.cs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            Type = AchievementType.Add,
+            TaskCount = 1
+        };
+
+        var taskDTO = new TaskDTO
+        {
+            Title = title,
+            Date = DateTime.Now,
+            Status = Domain.Enums.TaskStatus.ToDo,
+            UserId = userId
```
Comment (kalachck) [inline CatNote.IntegrationTests/TaskTests.cs:46]: Remove data creation into separate class

File: CatNote.IntegrationTests/UserTests.cs [hunk 4]
```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
```
Comment (kalachck) [inline CatNote.IntegrationTests/UserTests.cs:3]: Usings!

File: CatNote.IntegrationTests/UserTests.cs [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        };
+
+        userDTO1.Achievements = new List<AchievementDTO>
+        {
+            achievement1
+        };
+        userDTO1.Tasks = new List<TaskDTO>
+        {
+            task1
+        };
```
Comment (kalachck) [inline CatNote.IntegrationTests/UserTests.cs:63]: The same about data creation

**Review by kalachck — CHANGES_REQUESTED** [review]

File: CatNote.BLL/Services/AchievementService.cs [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+        return achievements;
+    }
+
+    public async Task CheckAchievementToAdd(int userId, CancellationToken cancellationToken)
+    {
+        var user = await GetUserModelById(userId, cancellationToken);
+
+        if (user == null || user.Tasks == null)
+            return;
```
[the remark this answers is not in the document]
Comment (MaliyOna) [reply CatNote.BLL/Services/AchievementService.cs:56]: for this part
image

[the remark this answers is not in the document]
Comment (kalachck) [reply CatNote.BLL/Services/AchievementService.cs:49]: I mean, If it's null just return false or throw exception. How to understand what is going on whan front send a request and achive nothing from back

File: CatNote.IntegrationTests/UserTests.cs [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+        {
+            AchievementData.AchievementAddDTOWithPoint(achievementId2, 1)
+        };
+
+        userDTO2.Tasks = new List<TaskDTO>
+        {
+            TaskData.TaskDTO(taskId2)
+        };
+
+        //Act
```
Comment (kalachck) [inline CatNote.IntegrationTests/UserTests.cs:58]: Resolve please unresolved comments

**Review by kalachck — CHANGES_REQUESTED** [review]

**Review by kalachck — APPROVED** [review]

**Review by Finka95 — APPROVED** [review]

**Pull request merged**